drgn/helpers/linux/kernel_config.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Kconfig
+------
+
+
+The ``drgn.helpers.linux.kconfig`` module provides helpers for reading kernel
+configuration from memory. This is only supported if CONFIG_IKCONFIG is set.
+"""
+
+
+import sys
+import gzip
+import types
+
+from drgn import NULL, cast, container_of
+
+__all__ = (
+    "kconfig",
+)
+
+def kconfig(prog):
+    try:
+        return prog.cache["kconfig_options_map"]
+    except KeyError:
+        pass
+
+    try:
+        start = prog.symbol("kernel_config_data").address
+        end = prog.symbol("kernel_config_data_end").address
+    except LookupError as e:
+        print("Kernel configuration data not found. Ensure your kernel is "
+              "compiled with CONFIG_IKCONFIG set.", file=sys.stderr)
+        raise e
```

Comment:
I consider the helpers library code, so rather than printing, this information should go in an exception:

```
except LookupError:
    raise LookupError("kernel configuration data not found; kernel must be compiled with CONFIG_IKCONFIG")
```